distributed/system.py

```diff
@@ -63,4 +65,25 @@ def memory_limit() -> int:
     return limit
 
 
+def process_memory(proc: psutil.Process | int | None = None) -> int:
+    """Return total memory used by a process
+
+    Parameters
+    ----------
+    proc: psutil.Process | int, optional
+        Process or PID to measure. Default: current process
+    """
+    if proc is None:
+        proc = psutil.Process()
+    elif isinstance(proc, int):
+        proc = psutil.Process(proc)
+
+    if LINUX:
+        minfo = proc.memory_full_info()
+        return minfo.rss + minfo.swap
```

Comment:
I don't understand why we can add this. swap is defined as the data that is swaped to disk

`swap (Linux): amount of memory that has been swapped out to disk.`

and is not actually in memory

Reply:
We measure "process" memory for these purposes:

- Calculate unmanaged memory (bokeh + prometheus + AMM optimistic memory heuristics). If you don't add swap, then managed memory may exceed rss. You will also see be misled into thinking that your host mounts enough physical RAM, whereas it could benefit from more.
- terminate threshold. If the user configured the memory_limit so that spilling is possible, then we should expect that the memory_limit was in fact set to physical memory + part of the swap file. If you omit swap from your measure, then you may end up never hitting the terminate threshold while the swap mount fills up. When it's fully saturated, the whole host will hang.
- In addition to the previous point: it's entirely legitimate and performant to disable target, spill, and pause thresholds, leaving everything to the os swapping mechanism, and just leave the terminate threshold on.
- To appreciate how much memory you reclaimed aftr gc and after spilling. Again, you have no control if some of the memory was swapped out or not. As a matter of fact, if you have both spill and swap systems going on, it's very likely that dask will choose to spill memory that was swapped out to begin with - because both algorithms are LRU(ish). This is obviously poorly performant, but a very realistic use case on dev boxes where you have a web browser and your IDE running at least.
- To notice memory leaks in tests. Again, if you're leaking you might be leaking swapped out memory.